docs/bugs.json

```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2022-7",
+        "name": "StorageWriteRemovalBeforeConditionalTermination",
+        "summary": "Calling Yul functions that conditionally terminate the external EVM call using ``return(...)`` or ``stop()`` may result in incorrect removals of storage writes.",
+        "description": "A call to a Yul function that conditionally terminates the external EVM call did not correctly mark all prior changes to storage as *used*, which may result in storage writes being incorrectly removed. This happens in cases in which it would have been valid to remove the store, if the Yul function in question never actually terminated, but control flow always returned back to the caller. Conditional termination within the same Yul block is not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may be incorrectly removed, whenever it would have been valid to remove the change without the ``return(...)`` or ``stop()``. In legacy code generation, only inline assembly that does not refer any Solidity variables and that involves conditionally-terminating user-defined assembly functions can be affected.",
```

Comment:
Past tense?
```suggestion
        "description": "A call to a Yul function that conditionally terminates the external EVM call did not correctly mark all prior changes to storage as *used*, which may result in storage writes being incorrectly removed. This used to happen in cases in which it would have been valid to remove the store, if the Yul function in question never actually terminated, and the control flow always returned back to the caller. Conditional termination within the same Yul block was not affected. In Solidity with via-IR code generation, any state change before a function conditionally calling ``return(...)`` or ``stop()`` in inline assembly, may have been incorrectly removed, whenever it would have been valid to remove the change without the ``return(...)`` or ``stop()``. In legacy code generation, only inline assembly that did not refer to any Solidity variables and that involveed conditionally-terminating user-defined assembly functions could be affected."
```